src/components/SignUp/SignUp.tsx

```diff
@@ -171,14 +206,15 @@ export class SignUp extends React.Component {
           password: this.state.passwordField.value,
         })
         .then((data) => {
-          const userData = {
-            username: data.username,
+          const userData: ILoggedInUser = {
```

Comment:
I thought fixing the shape of this and calling loginSuccess would've fixed the problem, but unfortunately not. Will try to dig a bit more on Monday.